### DISCUSSION TITLE
observation about running a python script

### DISCUSSION THREAD

**Discussion by peterrjohnston** [SCIInstitute/SCIRun#2275] [Q&A]
Hello,

I have been running some python scripts under scirun5 (Mac OS 11.3, SCIRun5 beta Y + Qt5.14.1) using the commandline command

% scirun -x -S script.py

I was wondering why it was necessary for a blank scirun window to appear during the execution of the script. This appears to serve no purpose as it is just a blank white window.

My main issue is that when I run these scripts as part of a larger script file, I cannot use the computer for fear of affecting or interrupting this window using a keyboard command.

Any thoughts would be most welcome.

Thanks,

Peter.

**Comment by dcwhite** [comment]
There are definitely some issues with window management and application lifetime in “quasi-headless” mode which is how you’re using it (the -x flag). I will look into those this milestone. As a workaround, you can build SCIRun as a completely headless app, with console-only interaction,  and that will take care of the spurious windows appearing while running scripts. The idea is you can fine-tune your scripts in GUI mode, then when it’s time to run them in production, use the full headless version which can be run safely in the background or on a remote server. The way to do this is to make a brand new binary folder, and turn on BUILD_HEADLESS in CMake at the Superbuild level. Once that is built, you can run scripts the same way with the -s/-S flag, or run interactively with a python shell using -i. This method is still somewhat experimental, so you might run into different issues, but please let me know if you do.

**Reply by peterrjohnston** [reply]
Dear Dan,

Thanks for your suggestion. I did this out, but the compilation failed. I guess this is one of those issues related to not having the correct c compiler. I have attached the log file from the build attempt and would be grateful if you could offer some further suggestions. I am running mac os 11.3.1.

If you require anything further, please let me know.

Thanks,

Peter.
[build.log](https://github.com/SCIInstitute/SCIRun/files/6490353/build.log)

**Reply by dcwhite** [reply]
Are you on an M1 machine? Either way, my current pull request has some fixes for build errors on M1 and probably 11.3 Intel as well. On M1, the Qt interface is not building yet. But once https://github.com/SCIInstitute/SCIRun/pull/2285 is merged (should be soon), a headless build on 11.3 ought to work.
